core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/Main.java

```diff
@@ -34,16 +55,77 @@ public static void main(String... args) throws Throwable {
             Performance.start();
             //TODO Move property key to -spi
             System.setProperty(BootstrapProperties.IS_UBERJAR, Boolean.TRUE.toString());
+
+            String processFile = System.getProperty(MAIN_PROCESS_FILE);
+
+            if (processFile != null) {
+                shutdownService = Executors.newSingleThreadExecutor();
+                shutdownService.submit(() -> {
+                        File uuidFile = new File(processFile);
+                        try {
+                            File watchedDir = uuidFile.getParentFile();
+                            register(watchedDir);
+                            processEvents(watchedDir, uuidFile.toPath());
```

Comment:
I'd inline the `register` method, it's just 2 lines, and mainly, the `processEvents` should be called something like `waitForFileDeleted` or something like that.